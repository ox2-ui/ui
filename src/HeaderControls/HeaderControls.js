import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

/**
 * HeaderControls Component
 */
const HeaderControls = props => {
  const {
    className: classNameProp,
    style,
    children,
  } = props;

  const className = classNames(
    'padding-x:7 padding-b:7 border-color:brand border:bottom border:2',
    classNameProp,
  );

  return (
    <div className={className} style={style}>
      {children}
    </div>
  );
};

HeaderControls.propTypes = {
  children: PropTypes.node,
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * Override the inline-styles of the root element.
   */
  style: PropTypes.object,
};

export default HeaderControls;
