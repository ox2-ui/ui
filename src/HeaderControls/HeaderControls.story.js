import React from 'react';
import { storiesOf } from '@kadira/storybook';
import HeaderControls from './HeaderControls';

storiesOf('HeaderControls', module)
  .add('default', () => (
    <HeaderControls>
      <div>sample text</div>
    </HeaderControls>
  ));

