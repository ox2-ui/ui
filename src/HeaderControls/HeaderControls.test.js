/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import HeaderControls from './HeaderControls';

it('renders correctly', () => {
  const wrapper = mount(
    <HeaderControls >
      <div>sample text</div>
    </HeaderControls>
  );

  expect(wrapper).toMatchSnapshot();
});
