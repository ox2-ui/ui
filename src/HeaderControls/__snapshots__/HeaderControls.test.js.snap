// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<HeaderControls
  className=""
  style={Object {}}
>
  <div
    className="padding-x:7 padding-b:7 border-color:brand border:bottom border:2"
    style={Object {}}
  >
    <div>
      sample text
    </div>
  </div>
</HeaderControls>
`;
